hnix-store-remote/src/System/Nix/Store/Remote.hs

```diff
@@ -156,6 +158,23 @@ buildPaths ps bm = do
     putPaths ps
     putInt $ fromEnum bm
 
+buildDerivation :: StorePath
+                -> Derivation StorePath Text
+                -> BuildMode
+                -> MonadStore BuildResult
+buildDerivation p drv buildMode = do
+  runOpArgs BuildDerivation $ do
+    putPath p
+    putDerivation drv
+    putEnum buildMode
+    -- XXX: reason for this is unknown
+    -- but without it protocol just hangs waiting for
+    -- more data. Needs investigation
+    putInt 0
```

Comment:
Only thing I could see is that libstore has a flush() operation. But possibly it's just putEnum that is not up to the task.

Reply:
You might be right about `putEnum` since it's the only use of it.

Reply:
Doesn't make much sense either, `putEnum` is just `putInt . fromEnum` so it's more like missing field. I don't care that much at this point as I'm playing with local store and daemon impl. so I'll be able to end-to-end test this without nix-daemon and the reason will probably become clear.